### PR TITLE
Monthly screening report ui implemented

### PR DESCRIPTION
**Pull request by siddh1004**
https://app.shortcut.com/simpledotorg/story/9561/create-screening-form-screen

### CODE REVIEW

File: app/src/test/java/org/simple/clinic/monthlyscreeningreports/form/QuestionnaireEntryUpdateTest.kt [hunk 1]
```diff
@@ -45,6 +45,30 @@ class QuestionnaireEntryUpdateTest {
...
+        .given(defaultModel)
+        .whenEvent(QuestionnaireResponseFetched(questionnaireResponse))
+        .then(assertThatNext(
+            hasModel(defaultModel.responseLoaded(questionnaireResponse)),
+            hasNoEffects()
+        ))
+  }
+
+  @Test
+  fun `when questionnaire response is saved, then go to complete screen`() {
```
Comment (prabhanshuguptagit) [inline app/src/test/java/org/simple/clinic/monthlyscreeningreports/form/QuestionnaireEntryUpdateTest.kt:62]: Is there a test like `when a questionnaire response is saved, submitted is set to "true", and the input values are saved inside "content"`? Should add integration tests if we haven't.

**Review by prabhanshuguptagit — CHANGES_REQUESTED** [review]
Looks good. Slightly unsure about the scope of this PR. Should we do a manual QA if there's any user facing changes?

File: app/src/main/java/org/simple/clinic/monthlyscreeningreports/complete/MonthlyScreeningReportCompleteEvent.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package org.simple.clinic.monthlyscreeningreports.complete
+
+import org.simple.clinic.widgets.UiEvent
+
+sealed class MonthlyScreeningReportCompleteEvent : UiEvent
+
+object DoneButtonClicked : MonthlyScreeningReportCompleteEvent() {
+  override val analyticsName = "Monthly Screening Reports Complete:Done Clicked"
```
[the remark this answers is not in the document]
Comment (siddh1004) [reply app/src/main/java/org/simple/clinic/monthlyscreeningreports/complete/MonthlyScreeningReportCompleteEvent.kt:8]: yes

File: app/src/main/java/org/simple/clinic/monthlyscreeningreports/complete/MonthlyScreeningReportCompleteScreen.kt [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  @Inject
+  lateinit var effectHandlerFactory: MonthlyScreeningReportCompleteEffectHandler.Factory
+
+  private val monthSubmittedTextView
+    get() = binding.monthSubmittedTextView
+
+  private val doneButton
+    get() = binding.doneButton
+
+  private val hardwareBackClicks = PublishSubject.create<Unit>()
```
[the remark this answers is not in the document]
Comment (siddh1004) [reply app/src/main/java/org/simple/clinic/monthlyscreeningreports/complete/MonthlyScreeningReportCompleteScreen.kt:52]: yes

File: app/src/main/java/org/simple/clinic/monthlyscreeningreports/complete/MonthlyScreeningReportCompleteScreen.kt [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    super.onAttach(context)
+    context.injector<Injector>().inject(this)
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    monthSubmittedTextView.text =
+        context?.resources?.getString(
+            R.string.monthly_screening_reports_submitted_with_date,
+            "October 2022"
```
[the remark this answers is not in the document]
Comment (siddh1004) [reply app/src/main/java/org/simple/clinic/monthlyscreeningreports/complete/MonthlyScreeningReportCompleteScreen.kt:87]: This is a test string. Same would be populated from questionnaire response.month key in the next PR.

File: app/src/main/java/org/simple/clinic/monthlyscreeningreports/form/QuestionnaireEntryScreen.kt [hunk 5]
```diff
@@ -156,6 +177,10 @@ class QuestionnaireEntryScreen : BaseScreen<
         .show()
   }
 
+  override fun goToMonthlyReportsCompleteScreen() {
+    router.push(MonthlyScreeningReportCompleteScreen.Key("October 2022"))
```
[the remark this answers is not in the document]
Comment (siddh1004) [reply app/src/main/java/org/simple/clinic/monthlyscreeningreports/form/QuestionnaireEntryScreen.kt:181]: This is a test string. Same would be populated from questionnaire response.month key in the next PR.

File: app/src/test/java/org/simple/clinic/monthlyscreeningreports/form/QuestionnaireEntryUpdateTest.kt (hunk 1 above)
Comment (siddh1004) [reply app/src/test/java/org/simple/clinic/monthlyscreeningreports/form/QuestionnaireEntryUpdateTest.kt:62]: Yes, it should be added

**Review by olttwa — APPROVED** [review]

**Review by olttwa — APPROVED** [review]

File: app/src/main/java/org/simple/clinic/monthlyscreeningreports/complete/MonthlyScreeningReportCompleteModel.kt [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package org.simple.clinic.monthlyscreeningreports.complete
+
+import android.os.Parcelable
+import kotlinx.parcelize.Parcelize
+
+@Parcelize
+data class MonthlyScreeningReportCompleteModel(
+    val month: String?
+) : Parcelable {
```
Comment (prabhanshuguptagit) [inline app/src/main/java/org/simple/clinic/monthlyscreeningreports/complete/MonthlyScreeningReportCompleteModel.kt:9]: `month_date`?

**Pull request merged**